src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,117 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final double GROW_CAPACITY_COEFFICIENT = 1.5;
+    private Object[] data;
+    private int size;
+
+    public ArrayList() {
+        this.data = new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size >= data.length) {
+            data = grow();
+        }
+        data[size] = value;
```

Comment:
```suggestion
        data[size++] = value;
```
you can do it like this